### PR TITLE
Ch firebase instrumentation tests #157084397

### PR DESCRIPTION
**Pull request by daumie**
#### What does this PR do?
- Moves  instrumentation tests from *Circle CI Build Server* to *Firebase*. 
- Slashes the time taken to perform instrumentation tests to approx 6 mins (Was 1approx 17mins initially)
- Gives a more  detailed report on instrumentation tests by including: _Test Results_ , _Logs_ , _Test Video_

#### Description of Task to be completed?
- Have the `instrumentation tests performed from Firebase.`
- Store test results on *GCP Storage Bucket*
- Send slack notifications when  tests  fail or pass

#### How should this be manually tested?
- When a push is made on any branch, the test reports are sent to the slack channel `#art-builds` with all the necessary information.
- Any interested party can click on the available buttons to get the indepth test  reports

image

#### Any background context you want to provide?
Instrumentation tests run on a device or an emulator. In the background, your app will be installed and then a testing app will also be installed which will control your app, launching it and running UI tests as needed.
Instrumentation tests can be used to test none UI logic as well. They are especially useful when you need to test code that has a dependency on a context.
Reducing the time taken for instrumentation tests improves feedback loop cycle to  developers increasing feedback productivity.

#### What are the relevant pivotal tracker stories?
https://www.pivotaltracker.com/story/show/157084397

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -374,6 +411,9 @@ workflows:
             - findbugs_lint
             - pmd_lint
             - checkstyle_lint
+      - instrumented_test:
+          requires:
+            - test
```
Comment (AnthonyGW) [inline .circleci/config.yml:414]: Maybe add a filter for only develop, staging and master branches

File: .circleci/config.yml [hunk 2]
```diff
@@ -193,43 +193,80 @@ jobs:
...
-      
+
       # Sending notification
       - run:
           name: Notifying slack channel on succeed
           when: on_success
           command: |
-            bash <(curl -s https://codecov.io/bash)
-            bash ~/repo/.circleci/notify_slack.sh
+            echo $CIRCLE_ARTIFACTS
```
Comment (AnthonyGW) [inline .circleci/config.yml]: Is there any purpose to these lines for `echo $CIRCLE_ARTIFACTS`?

**Review by AnthonyGW — APPROVED** [review]
I've checked the reports channel on slack. It works

File: .circleci/config.yml (hunk 1 above)
Comment (daumie) [reply .circleci/config.yml:414]: Thanks @AnthonyGW , I'm on it.

File: .circleci/config.yml (hunk 2 above)
Comment (daumie) [reply .circleci/config.yml]: I think that was erroneously added. I will be removing it in the next push.

**Review by MadgeKinyanjui — COMMENTED** [review]
@daumie what does the sample report sent on slack look like?

File: .circleci/config.yml (hunk 1 above)
Comment (daumie) [reply .circleci/config.yml:414]: I have added a filter to the suggested branches

File: .circleci/config.yml (hunk 2 above)
Comment (daumie) [reply .circleci/config.yml]: The lines have been removed. I think they were initially used for debugging.

**Pull request merged**